framework/di/Instance.php

```diff
@@ -128,7 +128,11 @@ public static function ensure($reference, $type = null, $container = null)
         }
 
         if ($reference instanceof self) {
-            $component = $reference->get($container);
+            try {
```

Comment:
there are two ways to evaluate:
1. choose by speed. In this case, the try catch is two times slower
2. choose by threshold: we define a duration threshold in which we evaluate. 

In this case we are talking about a difference of 0.005/100000 = 0.00000005 seconds per occurrence. Even if it is run 100 times, it will be 0.000005 seconds. Neglectable, so to speak. 

Of course, if it can be avoided, it's always good. Imagine 1000 of these usecases. Then it will be 0.005s which is already measurable by the debug toolbar.


![image](https://cloud.githubusercontent.com/assets/125726/20519231/69c613f4-b0a2-11e6-9e46-48b4eb059f93.png)

![image](https://cloud.githubusercontent.com/assets/125726/20519229/68721854-b0a2-11e6-8e87-e614e9051cb3.png)